packages/jest-preset-mc-app/fail-on-console.js

```diff
@@ -0,0 +1,123 @@
+const colors = require('colors/safe');
+
+const defaultErrorMessage = (methodName, bold) =>
+  `Expected test not to call ${bold(`console.${methodName}()`)}.\n\n` +
+  `If the ${methodName} is expected, test for it explicitly by mocking it out using ${bold(
+    'jest.spyOn'
+  )}(console, '${methodName}').mockImplementation() and test that the warning occurs.`;
+
+const failOnConsole = ({
+  errorMessage = defaultErrorMessage,
+  shouldFailOnAssert = false,
+  shouldFailOnDebug = false,
+  shouldFailOnLog = false,
+  shouldFailOnInfo = false,
+  shouldFailOnWarn = true,
+  shouldFailOnError = true,
+  skipTest,
+  silenceMessage,
+  logButNotThrowMessage,
+} = {}) => {
+  const flushUnexpectedConsoleCalls = (
+    methodName,
+    unexpectedConsoleCallStacks
+  ) => {
+    if (unexpectedConsoleCallStacks.length > 0) {
+      const messages = unexpectedConsoleCallStacks.map(([stack, message]) => {
+        const stackLines = stack.split('\n');
+        return (
+          `${colors.red(message)}\n` +
+          `${stackLines
+            .map((line, index) => {
+              if (index === stackLines.length - 1) {
+                return colors.white(line);
+              }
+              return colors.gray(line);
+            })
+            .join('\n')}`
+        );
+      });
+
+      const message = errorMessage(methodName, colors.bold);
+
+      throw new Error(`${message}\n\n${messages.join('\n\n')}`);
+    }
+  };
+
+  const patchConsoleMethod = (methodName) => {
+    let originalMethod = console[methodName];
+
+    const unexpectedConsoleCallStacks = [];
+
+    const captureMessage = (format, ...args) => {
+      const message = util.format(format, ...args);
+
+      if (silenceMessage && silenceMessage(message, methodName)) {
+        return;
+      }
+
+      if (logButNotThrowMessage && logButNotThrowMessage(message, methodName)) {
+        originalMethod(format, ...args);
+        return;
+      }
+
+      // Capture the call stack now so we can warn about it later.
+      // The call stack has helpful information for the test author.
+      // Don't throw yet though b'c it might be accidentally caught and suppressed.
+      const { stack } = new Error();
+      if (stack) {
+        unexpectedConsoleCallStacks.push([
+          stack.substr(stack.indexOf('\n') + 1),
+          message,
+        ]);
+      }
+    };
+
+    const newAssertMethod = (assertion, format, ...args) => {
+      if (assertion) {
+        return;
+      }
+
+      captureMessage(format, ...args);
+    };
+
+    const newMethod =
+      methodName === 'assert' ? newAssertMethod : captureMessage;
+
+    const canSkipTest = () => {
+      const currentTestState = expect.getState();
+      const testName = currentTestState.currentTestName;
+      const testPath = currentTestState.testPath;
+
+      if (skipTest && skipTest({ testName, testPath })) return true;
+
+      return false;
+    };
+    const shouldSkipTest = canSkipTest();
+
+    beforeEach(() => {
```

Comment:
Can you share how this skip test functionality would work?

Reply:
We don't use that but I kept it because it's in the original library. https://github.com/ValentinH/jest-fail-on-console#skiptest